indexer/src/main/java/au/org/aodn/esindexer/utils/CacheArdcVocabsUtils.java

```diff
@@ -0,0 +1,148 @@
+
+    @Autowired
+    ElasticsearchClient portalElasticsearchClient;
+
+    @Value("${elasticsearch.categories-index.name}")
+    String categoriesIndexName;
+
+    @Autowired
+    ObjectMapper objectMapper;
+
+
+    protected void indexingDiscoveryCategoriesIndex(List<CategoryVocabModel> categoryVocabModels) throws IOException {
+        BulkRequest.Builder bulkRequest = new BulkRequest.Builder();
+        // recreate index from mapping JSON file
+        elasticSearchIndexService.createIndexFromMappingJSONFile(AppConstants.AODN_DISCOVERY_PARAMETER_VOCABULARIES_MAPPING_JSON_FILE, categoriesIndexName);
+        log.info("Indexing all categoryVocabModel to {}", categoriesIndexName);
+        for (CategoryVocabModel categoryVocabModel : categoryVocabModels) {
+            try {
+                // convert categoryVocabModel values to binary data
+                log.debug("Ingested json is {}", objectMapper.writerWithDefaultPrettyPrinter().writeValueAsString(categoryVocabModel));
+                // send bulk request to Elasticsearch
+                bulkRequest.operations(op -> op
+                    .index(idx -> idx
+                        .index(categoriesIndexName)
+                        .document(categoryVocabModel)
+                    )
+                );
+            } catch (JsonProcessingException e) {
+                log.error("Failed to ingest categoryVocabModel to {}", categoriesIndexName);
+                throw new RuntimeException(e);
+            }
+        }
+
+        BulkResponse result = portalElasticsearchClient.bulk(bulkRequest.build());
+
+        // Flush after insert, otherwise you need to wait for next auto-refresh. It is
+        // especially a problem with autotest, where assert happens very fast.
+        portalElasticsearchClient.indices().refresh();
+
+        // Log errors, if any
+        if (result.errors()) {
+            log.error("Bulk had errors");
+            for (BulkResponseItem item: result.items()) {
+                if (item.error() != null) {
+                    log.error("{} {}", item.error().reason(), item.error().causedBy());
+                }
+            }
+        } else {
+            log.info("Finished bulk indexing categoryVocabModel items to index: " + categoriesIndexName);
+        }
+        log.info("Total documents in index: " + categoriesIndexName + " is " + elasticSearchIndexService.getDocumentsCount(categoriesIndexName));
+    }
+
+
+    /*
+    fetch vocabularies from ARDC and index to discovery categories index once the bean is created
+     */
+    @PostConstruct
+    public void refreshDiscoveryCategoriesIndex() throws IOException {
+        log.info("Fetching AODN Discovery Parameter Vocabularies from ARDC");
+        List<CategoryVocabModel> categoryVocabModels = ardcVocabsService.getParameterCategory(vocabApi);
+        indexingDiscoveryCategoriesIndex(categoryVocabModels);
+    }
+
+    @Cacheable(AppConstants.AODN_DISCOVERY_CATEGORIES_CACHE)
```

Comment:
Add TODO comment to remind how to avoid multiple refresh on multiple indexer instance?

Reply:
surething